### PR TITLE
New user - form

### PR DESCRIPTION
**Pull request by sebastianmusial**
* Simple form based on model fields without connection with API.
* Box component - clean up 
* FIltration - visual changes in component

**@TODO:**
* Form validation- error and success information.
* Tests

zrzut ekranu 2018-01-21 o 01 14 02

zrzut ekranu 2018-01-23 o 20 24 16

### CODE REVIEW

File: src/app/+models/containers/model-new/model-new.component.ts [hunk 1]
```diff
@@ -1,25 +1,55 @@
...
+import { Subject } from 'rxjs/Subject';
+import { FormBuilder, FormGroup } from '@angular/forms';
+import { AutoUnsubscribe } from 'ngx-auto-unsubscribe';
 import { Store } from '@ngrx/store';
-import { ModelState, modelSelectors } from '../../store';
+import { State, selectors } from '../../../store';
+import { map, tap } from 'rxjs/operators';
+import { MetadataField } from '../../../metadata/interfaces';
+import { modelActions } from '../../store';
+import { NewModel } from '../../interfaces/index';
```
Comment (JozefFlakus) [inline src/app/+models/containers/model-new/model-new.component.ts]: `import { NewModel } from '../../interfaces';`

File: src/app/+models/containers/model-new/model-new.component.ts [hunk 2]
```diff
@@ -1,25 +1,55 @@
...
+import { transformToFormFields } from '../../shared/index';
```
Comment (JozefFlakus) [inline src/app/+models/containers/model-new/model-new.component.ts]: `import { transformToFormFields } from '../../shared/';`

File: src/app/+models/shared/model.repository.ts [hunk 3]
```diff
@@ -20,4 +21,10 @@ export class ModelRepository {
       .get(environment.apiUrl + `${modelName}/${id}`)
       .pipe(catchError(error => Observable.throw(error || 'Server error')));
   }
+
+  addSingle(modelName: string, body: NewModel) {
```
Comment (JozefFlakus) [inline src/app/+models/shared/model.repository.ts]: `addSingle` -> `postSingle` (this way we will know that this is a POST call)

**Comment by JozefFlakus** [comment]
@sebastianmusial should I change param `field` to `name` in `@baristaJS/core`?

**Comment by sebastianmusial** [comment]
@JozefFlakus Yes, if it's no problem.

**Comment by JozefFlakus** [comment]
@sebastianmusial I will try to change it in the next `@barista/core` PR - after that please make proper updates in `@barista/admin`

File: src/app/+models/containers/model-new/model-new.component.html [hunk 4]
```diff
@@ -15,13 +15,13 @@ <h1>
         <b-model-field
           *ngFor="let field of this.fields"
         >
-          <mat-form-field>
+          <mat-form-field *ngIf="field.inputType !== 'checkbox'">
```
Comment (JozefFlakus) [inline src/app/+models/containers/model-new/model-new.component.html]: maybe https://angular.io/api/common/NgSwitch

File: src/app/+models/containers/model-new/model-new.component.ts [hunk 5]
```diff
@@ -43,9 +43,12 @@ export class ModelNewComponent implements OnInit, OnDestroy {
       );
 
     this.submit$
-      .subscribe(() => this.store.dispatch(
-        new modelActions.ModelSingleAdd(this.modelName, { data: this.form.value }))
-      );
+      .subscribe(() => {
+        console.log(this.form.value);
```
Comment (JozefFlakus) [inline src/app/+models/containers/model-new/model-new.component.ts]: add `// @TODO`

File: src/app/+models/components/model-field/model-field.component.html [hunk 6]
```diff
@@ -1,3 +1,52 @@
...
+  <ng-container [ngSwitch]="field.inputType">
+
+    <!-- Checkbox -->
+    <div
+      class="model-field__item model-field__checkbox"
+      *ngSwitchCase="'checkbox'"
+    >
+      <mat-checkbox
+        class="model-field__item"
+        formControlName="{{ field.name }}"
```
Comment (JozefFlakus) [inline src/app/+models/components/model-field/model-field.component.html]: `[formControlName]="field.name"`

File: src/app/+models/components/model-field/model-field.component.html [hunk 7]
```diff
@@ -1,3 +1,52 @@
...
+        required="{{ field.isRequired }}"
```
Comment (JozefFlakus) [inline src/app/+models/components/model-field/model-field.component.html]: [required]="field.isRequired"

File: src/app/+models/store/model.effects.spec.ts [hunk 8]
```diff
@@ -70,4 +75,32 @@ describe('ModelEffects', () => {
 
     expect(effects.getModelCollection$).toBeObservable(expected$);
   });
+
+  it(`postSingleModel$ dispatches ${MODEL_SINGLE_ADD_SUCCESS}`, () => {
+    const mockedResponse = [{ id: 1 }, { id: 2 }];
```
Comment (JozefFlakus) [inline src/app/+models/store/model.effects.spec.ts]: It will work, but be aware that real use case shouldn't contain an array but one object

**Pull request merged**